src/Hintman/Server/Auth.hs

```diff
@@ -21,11 +28,37 @@ type instance AuthServerData (AuthProtect "GitHub") = ()
 type HintmanAuthContextHandlers = '[AuthHandler Request ()]
 type HintmanAuthContext = Context HintmanAuthContextHandlers
 
+verifySignature :: Request -> Handler ()
+verifySignature req = do
+    body <- liftIO $ requestBody req
+    verifySignature' body header
+  where
+    header :: Maybe ByteString
+    header = lookup "x-hub-signature" (requestHeaders req)
+
+verifySignature' :: ByteString -> Maybe ByteString -> Handler ()
+verifySignature' _ Nothing = throwError $ err401 { errBody = "Empty signature." }
+verifySignature' payload (Just signature) = do
+    let (_, digest) = breakSubstring signature "="
```

Comment:
I misread the example provided in the issue, the first argument (tuple) should not be discarded, it signals which algorithm to use and it might not always be Sha1!

Reply:
Let's assume `sha1` for now and figure out something else later